File: TodoApp-SpringBoot/todoApp-using-ReactNSpringboot/src/main/java/com/fullstack/todo/service/TodoService.java
package com.fullstack.todo.service;

import com.fullstack.todo.dto.TodoDTO;

import java.util.List;

public interface TodoService {

//    Create Todo
    TodoDTO addTodo(TodoDTO todoDTO);

//    GET TODO
    TodoDTO getTodo(Long id);

//    GET All Todos
    List<TodoDTO> getAllTodos();

//    Update Todos
    TodoDTO updateTodo(TodoDTO todoDTO, Long id);

//    DELETE Todo
    void deleteTodo(Long id);

//    Completed Todo
    TodoDTO completeTodo(Long id);

//    InComplete Todo
    TodoDTO inCompleteTodo(Long id);
}

File: TodoApp-SpringBoot/todo-ui/src/services/TodoService.js
// this is Service class which integrates with backedn API for REST API

import axios from "axios";
import { getToken } from "./AuthService";

const URL = "http://localhost:8080/api/todos";

// Add a request interceptor
// axios.interceptors.request.use(
//   function (config) {
//     config.headers["Authorization"] = getToken();
//     return config;
//   },
//   function (error) {
//     // Do something with request error
//     return Promise.reject(error);
//   }
// );

// getAll Todos
export const getAllTodos = () => axios.get(URL);

// addTodo to db
export const saveTodo = (todo) => axios.post(URL, todo);

// GETTodo REST API call
export const getTodo = (id) => axios.get(URL + "/" + id);

// UPDATE TODO
export const updateTodo = (id, todo) => axios.put(URL + "/" + id, todo);

// DELETE REST CLient API call
export const deleteTodo = (id) => axios.delete(URL + "/" + id);

// Patch
export const completedTodo = (id) => axios.patch(URL + "/" + id + "/complete");

// Patch
export const inCompletedTodo = (id) =>
  axios.patch(URL + "/" + id + "/in-complete");
